src/Compilers/CSharp/Portable/Syntax/SyntaxEquivalence.cs

```diff
@@ -69,16 +66,24 @@ private static bool AreTokensEquivalent(GreenNode before, GreenNode after)
             switch ((SyntaxKind)before.RawKind)
             {
                 case SyntaxKind.IdentifierToken:
-                    return ((Green.SyntaxToken)before).ValueText == ((Green.SyntaxToken)after).ValueText;
+                    if (((Green.SyntaxToken)before).ValueText != ((Green.SyntaxToken)after).ValueText)
+                    {
+                        return false;
+                    }
+                    break;
 
                 case SyntaxKind.NumericLiteralToken:
                 case SyntaxKind.CharacterLiteralToken:
                 case SyntaxKind.StringLiteralToken:
                 case SyntaxKind.InterpolatedStringTextToken:
-                    return ((Green.SyntaxToken)before).Text == ((Green.SyntaxToken)after).Text;
+                    if (((Green.SyntaxToken)before).Text != ((Green.SyntaxToken)after).Text)
+                    {
+                        return false;
+                    }
+                    break;
             }
 
-            return true;
+            return AreNullableDirectivesEquivalent(before, after, ignoreChildNode);
```

Comment:
so this is called on every token?

Reply:
Yes

---
In reply to: [333780086](https://github.com/dotnet/roslyn/pull/39227#discussion_r333780086) [](ancestors = 333780086)

Reply:
that seems super unfortunate.  could we have a bit that says if the tree contains a nullable directive in it?  note: the tree *already* says if it contains directives.  so as long as we're checking that, then i think i'm fine.

Reply:
@CyrusNajmabadi `GetDirectives()` has a fast path that returns an empty list if the tree does not contain directives. Is this what you're looking for, are you looking for another bit indicating nullable directives explicitly?

Reply:
> @CyrusNajmabadi GetDirectives() has a fast path that returns an empty list if the tree does not contain directives. Is this what you're looking for

Yes, that's good enough for me.

> are you looking for another bit indicating nullable directives explicitly?

I think that would also likely be a good idea given how necessary it will be to ask for this specific directive.  But i'm nto going to block on that. :)